app/2.0/docs/devguide/cross-scope-styling.md

```diff
@@ -0,0 +1,331 @@
+element encapsulated in `my-toolbar`'s internal implementation.  In the
+future the `my-toolbar` author can rename the `title` class or
+restructure the internal details of `my-toolbar` without changing the custom
+property exposed to users.
+
+You can also include a default value in the `var()` function, to use in case the user
+doesn't set the custom property:
+
+```css
+color: var(--my-toolbar-title-color, blue);
+```
+
+Thus, custom CSS properties are a powerful way for element authors to
+expose a theming API to their users in a way that naturally fits right alongside
+normal CSS styling. 
+
+### Custom CSS mixins
+
+It may be tedious (or impossible) for an element author to predict every
+CSS property that may be important for theming, let alone expose every
+property individually.
+
+An element author can define a set of CSS properties as a single custom property and
+then allow all properties in the set to be applied to a specific CSS rule
+in an element's local DOM. The extension enables this with a mixin capability
+that is analogous to `var`, but which allows an entire set of properties
+to be mixed in.
+
+Use `@apply` to apply a mixin:
+
+<pre><code class="language-js">@apply(--<var>mixin-name</var>);</code></pre>
+
+Defining a mixin is just like defining a custom property, but the
+value is an object that defines one or more rules:
+
+<pre><code class="language-css"><var>selector</var> {
+  --<var>mixin-name</var>: {
+    /* rules */
+  };
+}</code></pre>
+
+Example: { .caption }
+
+```html
+<dom-module id="my-toolbar">
+  <template>
+    <style>
+      :host {
+        padding: 4px;
+        background-color: gray;
+        /* apply a mixin */
+        @apply(--my-toolbar-theme);
+      }
+      .title {
+        @apply(--my-toolbar-title-theme);
+      }
+    </style>
+    <span class="title">{{title}}</span>
+  </template>
+  ...
+</dom-module>
+```
+
+Example usage of `my-toolbar`: { .caption }
+
+```html
+<dom-module id="my-element">
+  <template>
+    <style>
+      /* Apply custom theme to toolbars */
+      :host {
+        --my-toolbar-theme: {
+          background-color: green;
+          border-radius: 4px;
+          border: 1px solid gray;
+        };
+        --my-toolbar-title-theme: {
+          color: green;
+        };
+      }
+      /* Make only toolbars with the .warning class red and bold */
+      .warning {
+        --my-toolbar-title-theme: {
+          color: red;
+          font-weight: bold;
+        };
+      }
+    </style>
+    <my-toolbar title="This one is green."></my-toolbar>
+    <my-toolbar title="This one is green too."></my-toolbar>
+    <my-toolbar class="warning" title="This one is red."></my-toolbar>
+  </template>
+
+  <script>
+    Polymer({ is: 'my-element'});
+  </script>
+</dom-module>
+```
+
+## Inheritance
+
+If an element doesn't override styling information, that element inherits styles from its parent:
+
+```
+<style is="custom-style">
```

Comment:
This should be wrapped in a `<custom-style>`
Also, should probably show the import for the custom-style element.